source/common/upstream/upstream_impl.cc

```diff
@@ -643,12 +643,24 @@ ClusterInfoImpl::ClusterInfoImpl(
                       envoy::api::v2::Cluster_LbPolicy_Name(config.lb_policy()),
                       envoy::api::v2::Cluster_DiscoveryType_Name(config.type())));
     }
+    if (config.has_lb_subset_config()) {
+      throw EnvoyException(
```

Comment:
Turns out this isn't reachable because of the check in [original_dst_cluster.cc](https://github.com/envoyproxy/envoy/blob/master/source/common/upstream/original_dst_cluster.cc#L187). 

From the code/comments it seemed the plan was for the original dst cluster to become a custom cluster type to be used in conjunction with `cluster_provided` lb policy. When that happens I think this case become redundant and will get removed.

That said, I could remove the subset lb check in original_dst_cluster.cc in favor of this one.

Reply:
Yeah, I'd either suggest a TODO for you to pick up the change which clean this up, or a refactor now, rather than have redundant code in here indefinitely.

Reply:
I remove the check in the original dst LB.